src/pybind/mgr/cephadm/module.py

```diff
@@ -2058,7 +2059,12 @@ def matches_network(host):
             self.log.debug('Placing %s.%s on host %s' % (
                 daemon_type, daemon_id, host))
 
-            self.cephadm_services[daemon_type].create(daemon_spec)
+            try:
+                self.cephadm_services[daemon_type].create(daemon_spec)
+            except (RuntimeError, OrchestratorError) as e:
+                self.log.warning("Failed while placing %s.%s on %s, skipping:\n%s" % (daemon_type, daemon_id, host, e))
```

Comment:
Users really fail to look into the logs when something goes wrong. Let's use our new fancy events for that:
```python
                self.events.for_service(spec, 'ERROR', f"Failed while placing {daemon_type}.{daemon_id} on {host}: {e}")
```

Reply:
Sounds good

Reply:
uncomfortable with the f-string, though; that's 3.6+ only, right?  Can we assume that much Py3 yet?

Reply:
mgr/cephadm/module.py is py36+ We live in the fluffy warm container with a very predictable setup.

Reply:
I thought the mon and mgr were the two exceptions for bootstrapping a cluster (because something has to run the orchestration to set up containerized daemons), so they're running native, at least temporarily, no?

Reply:
`src/pybind/mgr/cephadm` is never executed outside of a container.